File: Uberly-backend/src/main/java/Elena/Uberly_backend/Entity/Reaction.java
package Elena.Uberly_backend.Entity;

import Elena.Uberly_backend.Enum.ReactionType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIncludeProperties;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "reactions")
public class Reaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Enumerated(EnumType.STRING)
    private ReactionType type;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    @JsonIncludeProperties (value = {"id", "username", "pictureProfile"})
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "comment_id")
    @JsonIgnore
    private Comment comment;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "feedback_id")
    @JsonIgnore
    private Feedback feedback;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id")
    @JsonIgnore
    private Post post;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reply_id")
    @JsonIgnore
    private Reply reply;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "meme_id")
    @JsonIgnore
    private Meme meme;
}

File: Uberly-frontend/src/app/interfaces/reaction.ts
import { Comment } from './comment';
import { Feedbacks } from './feedbacks';
import { Meme } from './meme';
import { Post } from './post';
import { Reactiontypes } from './reactiontypes';
import { Reply } from './reply';
import { User } from './user';

export interface Reaction {
  id?: number;
  userId: number;
  type: Reactiontypes;
  comment?: Comment;
  feedback?: Feedbacks;
  postId?: number;
  reply?: Reply;
  meme?: Meme;
}
